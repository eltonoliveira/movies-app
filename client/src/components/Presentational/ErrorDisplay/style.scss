@import "sass/app.scss";

.error-display{
  margin-top: 50px;
  .error-display-info{
    display: block;
  }
}
