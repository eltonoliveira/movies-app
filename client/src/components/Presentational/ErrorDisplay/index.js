import React from 'react';
import './style.scss';

const errorImg = require('./img/error.gif');

const ErrorDisplay = () => {
  return (
    <div className="error-display">
      <img alt="" src={errorImg} />
      <span className="error-display-info">
        Something wen wrong please try again later.
      </span>
    </div>
  );
};

export default ErrorDisplay;
