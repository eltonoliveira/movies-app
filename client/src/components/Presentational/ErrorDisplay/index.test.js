import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import ErrorDisplay from './index';

configure({adapter: new Adapter()});


describe('tests the error component', () => {
    test('shall render the error display without error', () => {
        const wrapper = shallow(<ErrorDisplay/>);
        expect(wrapper).toMatchSnapshot();
    });
})
