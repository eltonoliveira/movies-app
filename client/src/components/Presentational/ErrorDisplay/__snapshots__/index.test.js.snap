// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the error component shall render the error display without error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorDisplay />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="error.gif"
        />,
        <span
          className="error-display-info"
        >
          Something wen wrong please try again later.
        </span>,
      ],
      "className": "error-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "error.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Something wen wrong please try again later.",
          "className": "error-display-info",
        },
        "ref": null,
        "rendered": "Something wen wrong please try again later.",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="error.gif"
          />,
          <span
            className="error-display-info"
          >
            Something wen wrong please try again later.
          </span>,
        ],
        "className": "error-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "error.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something wen wrong please try again later.",
            "className": "error-display-info",
          },
          "ref": null,
          "rendered": "Something wen wrong please try again later.",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
